### ISSUE TITLE
Never uses TAN even if tan_insert is True

### ISSUE THREAD

**Issue by hogliux** [grindsa/dkb-robo#46]
Thank you for this great project. This is exactly what I need!!

Unfortunately, no matter if `tan_insert` is `True` or `False`, dkb-robo will always ask for confirmation via the banking app. For example:

```sh
python3 dkb_robo/cli.py --debug -u xxxxx -p xxx --use-tan accounts 
DKBRobo._login()

DKBRobo._new_session()

Starting new HTTPS connection (1): banking.dkb.de:443
https://banking.dkb.de:443 "GET /login HTTP/1.1" 200 None
DKBRobo._get_token()

https://banking.dkb.de:443 "POST /api/token HTTP/1.1" 200 None
DKBRobo._get_mfa_methods()

https://banking.dkb.de:443 "GET /api/mfa/mfa/methods?filter%5BmethodType%5D=seal_one HTTP/1.1" 200 None
_select_mfa_device()

Pick an authentication device from the below list:
[0] - DKB-App on XXXXXX
[1] - DKB-App on XXXXXX
[2] - DKB-App on XXXXXX
:0
```

The same occurs if I edit [line 27 of the `dkb_example.py`](https://github.com/grindsa/dkb-robo/blob/5beaa44bc763d6fcb1a454ee55ccd01aaf9b9d5d/doc/dkb_example.py#L27) as follows:

```python
    with DKBRobo(DKB_USER, DKB_PASSWORD, True, False) as dkb:
```

The same problem occurs.

**Comment by grindsa** [comment]
I can confirm that this is a bug. Chiptan requires the old legacy UI.

the following workaround should work:

```
with DKBRobo(dkb_user=DKB_USER, dkb_password=DKB_PASSWORD, tan_insert=True, legacy_login=True) as dkb:
    pprint(dkb.account_dic)
```

Will meanwhile work on a fix...

**Comment by hogliux** [comment]
Thank you so much for your quick response. However, when I use legacy UI, I get the following error:

```
DKBRobo._login()

DKBRobo._new_instance()

Starting new HTTPS connection (1): www.ib.dkb.de:443
https://www.ib.dkb.de:443 "GET /banking HTTP/1.1" 200 None
https://www.ib.dkb.de:443 "POST /banking HTTP/1.1" 200 None
DKBRobo._ctan_check()

Login failed: LinkNotFoundError
```

**Comment by grindsa** [comment]
Fixes got released with [v0.23.1](https://github.com/grindsa/dkb-robo/releases/tag/0.23.1).  Can you please check if you still get the `LinkNotFoundError` error?

**Comment by hogliux** [comment]
I'm still getting the `LinkNotFoundError`. I've modified your code to dump the html page, where soup is looking for the link. I've just invited you to a repo which contains this html page as I don't want to share this html page in the public (it may contain personal information or encoded session tokens etc.).

**Comment by grindsa** [comment]
This looks strange. Can you enable debugging as below and trigger the error and upload the debug log to the repo you create? Can you please furhter add the dkb_robo.py that I see at which time you dumped the html page?

```python
with DKBRobo(dkb_user=DKB_USER, dkb_password=DKB_PASSWORD, tan_insert=True, legacy_login=True, debug=True) as dkb:
    pprint(dkb.account_dic)
```

**Comment by hogliux** [comment]
This is the output:

```bash
Traceback (most recent call last):
  File "/home/XXXXX/Developer/dkb-robo/dkb_robo/dkb_robo.py", line 1733, in _legacy_login
    login_confirmed = self._ctan_check(soup)
                      ^^^^^^^^^^^^^^^^^^^^^^
  File "/home/XXXXX/Developer/dkb-robo/dkb_robo/dkb_robo.py", line 1348, in _ctan_check
    self.dkb_br.select_form('form[name="next"]')
  File "/usr/lib/python3/dist-packages/mechanicalsoup/stateful_browser.py", line 207, in select_form
    raise LinkNotFoundError()
mechanicalsoup.utils.LinkNotFoundError

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/XXXXX/Developer/dkb-robo/dkb_example.py", line 27, in <module>
    with DKBRobo(dkb_user=DKB_USER, dkb_password=DKB_PASSWORD, tan_insert=True, legacy_login=True, debug=True) as dkb:
  File "/home/XXXXX/Developer/dkb-robo/dkb_robo/dkb_robo.py", line 170, in __enter__
    self._legacy_login()
  File "/home/XXXXX/Developer/dkb-robo/dkb_robo/dkb_robo.py", line 1742, in _legacy_login
    raise DKBRoboError('Login failed: LinkNotFoundError') from err
dkb_robo.dkb_robo.DKBRoboError: Login failed: LinkNotFoundError
```

**Comment by hogliux** [comment]
Attached is my modified `dkb_robo.py`. See lines following 1711
[dkb_robo.py.txt](https://github.com/grindsa/dkb-robo/files/12479616/dkb_robo.py.txt)

**Comment by grindsa** [comment]
Thank you. I do not really understand why this happens. [This](https://github.com/grindsa/dkb-robo/blob/0af240373d266fecbe4e1060bc899df2e99fc679/dkb_robo/dkb_robo.py#L1347-L1351) is the code block causing issues.  It is true that the link canno tbe found, however the resulting exception. should be catched by the code and just throw an error in the logfile.

Anyway I do not think that this code block is needed anymore so I removed it.  Change has been pushed into a [separate branch](https://github.com/grindsa/dkb-robo/tree/tan_insert).  Can you please try if this fixes your issue (replacing dkb_robo.py should be enough)

**Comment by hogliux** [comment]
Yes it works now. Thank you!

**Comment by grindsa** [comment]
Fixed with [0.23.2](https://github.com/grindsa/dkb-robo/releases/tag/0.23.2)

**Comment by hogliux** [comment]
Thank you!
